### DISCUSSION TITLE
My code snippets not working on CodePen

### DISCUSSION THREAD

**Discussion by stackerito** [alpinejs/alpine#1869] [1. Help]
I want to ask others about problems in my code, but the snippet doesn't work on CodePen (Or Stack Overflow)

This is the pen: https://codepen.io/Stackerito/pen/gOWeYmL

the problem is that after you open the sidebar, you should be able to click the "Click" button to slide it off the screen (because it's supposed to open a new menu)

In my local dev environment it does slide (and opens empty menu), but on CodePen it won't open it when I click the "Click" button

**Comment by KevinBatdorf** [comment]
You forgot the `transform` class on that item.

You should also use a `button` element for that button, or otherwise handle all the same events.

**Reply by stackerito** [reply]
Thank you, the ```transform``` did help. But where should I use a ```button```? Wherever I put it, it breaks the entire span styling.
What do you mean handle all same events? Can you explain? I thought I can only handle the main wrapping div and it should work?

**Reply by KevinBatdorf** [reply]
Some elements in the browser have events applied to them by default for accessibility reasons. If you're not going to use a button element, you have a duty as a developer to cover all those events. For a button you have click and two keydown events you need to cover. More info here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

Better to use a button when adding `@click` events, and style it appropriately.

**Reply by stackerito** [reply]
You mean that I should convert all the content in the main ```<span>``` of each category in the menu to a ```<button>```?

For example this entire piece:

```
  <span class="flex p-4 border-b cursor-pointer">
                    <svg @click="toggle()" xmlns="http://www.w3.org/2000/svg" class="items-end h-5 w-5" viewBox="0 0 20 20" fill="currentColor">
                        <path fill-rule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clip-rule="evenodd" />
                    </svg>
                </span>
            </div>

            <div x-data="{ showMenu: true }" :class="showMenu ? '' : 'ease-in-out transition-all duration-300 z-20 -translate-x-full'">
                <span @click="showMenu = false" class="flex items-center p-4 hover:bg-indigo-500 hover:text-white cursor-pointer"><span class="mr-2">

                    </span>
                    <span>Click</span>
                    <span class="flex w-full justify-end">
                        <svg xmlns="http://www.w3.org/2000/svg" class="h-5 w-5" viewBox="0 0 20 20" fill="currentColor">
                            <path fill-rule="evenodd" d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z" clip-rule="evenodd" />
                        </svg>
                    </span>
                </span>
```


Because I tried to wrap the entire code in ```<button></button>``` but it completely broke the span styling. I tried to make it work but I'm a beginner so I couldn't find a way - could you please help me with this one?

**Reply by KevinBatdorf** [reply]
You can just change the `<span` to `<button` and move the `@click` event up to the button
